clusterloader2/pkg/measurement/common/bundle/test_metrics.go

```diff
@@ -0,0 +1,184 @@
+		glog.Errorf("TestMetrics: apiResponsiveness creation error: %v", err)
+	}
+	if metrics.etcdMetrics, err = measurement.CreateMeasurement("EtcdMetrics"); err != nil {
+		glog.Errorf("TestMetrics: etcdMetrics creation error: %v", err)
+	}
+	if metrics.metricsForE2E, err = measurement.CreateMeasurement("MetricsForE2E"); err != nil {
+		glog.Errorf("TestMetrics: metricsForE2E creation error: %v", err)
+	}
+	if metrics.apiserverCPUProfile, err = measurement.CreateMeasurement("CPUProfile"); err != nil {
+		glog.Errorf("TestMetrics: apiserverCPUProfile creation error: %v", err)
+	}
+	if metrics.apiserverMemoryProfile, err = measurement.CreateMeasurement("MemoryProfile"); err != nil {
+		glog.Errorf("TestMetrics: apiserverMemoryProfile creation error: %v", err)
+	}
+	if metrics.schedulerMemoryProfile, err = measurement.CreateMeasurement("MemoryProfile"); err != nil {
+		glog.Errorf("TestMetrics: schedulerMemoryProfile creation error: %v", err)
+	}
+	if metrics.resourceUsageSummary, err = measurement.CreateMeasurement("ResourceUsageSummary"); err != nil {
+		glog.Errorf("TestMetrics: resourceUsageSummary creation error: %v", err)
+	}
+	if metrics.schedulingMetrics, err = measurement.CreateMeasurement("SchedulingMetrics"); err != nil {
+		glog.Errorf("TestMetrics: schedulingMetrics creation error: %v", err)
+	}
+	return &metrics
+}
+
+type testMetrics struct {
+	apiResponsiveness      measurement.Measurement
+	etcdMetrics            measurement.Measurement
+	metricsForE2E          measurement.Measurement
+	apiserverCPUProfile    measurement.Measurement
+	apiserverMemoryProfile measurement.Measurement
+	schedulerMemoryProfile measurement.Measurement
+	resourceUsageSummary   measurement.Measurement
+	schedulingMetrics      measurement.Measurement
+}
+
+// Execute supports two actions. start - which sets up all metrics.
+// stop - which stops all metrics and collects all measurements.
+func (t *testMetrics) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+	var summaries []measurement.Summary
+	var errs []error
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return summaries, err
+	}
+
+	switch action {
+	case "start":
+		summary, err := execute(t.apiResponsiveness, config, map[string]interface{}{
+			"action": "reset",
+		})
+		appendResults(&summaries, &errs, summary, err)
+		summary, err = execute(t.etcdMetrics, config, map[string]interface{}{
+			"action": "start",
+		})
+		appendResults(&summaries, &errs, summary, err)
+		summary, err = execute(t.resourceUsageSummary, config, map[string]interface{}{
+			"action": "start",
+		})
+		appendResults(&summaries, &errs, summary, err)
+		summary, err = execute(t.schedulingMetrics, config, map[string]interface{}{
+			"action": "reset",
+		})
+		appendResults(&summaries, &errs, summary, err)
+	case "gather":
+		summary, err := execute(t.apiResponsiveness, config, map[string]interface{}{
```

Comment:
Define options once, instead of defining it for every call separately.

Reply:
The problem is that there are metrics that use the same key but with different values ("action": "start" | "action": "reset"), so it can't be defined once.

Reply:
Then define two or three or sth sets of those, e.g.

startParams, resetParams, and so on.

Reply:
Done.